test/test_autograd.py

```diff
@@ -886,6 +886,24 @@ def backward(ctx, grad_output):
         y = x.masked_fill(mask, 0)
         y.sum().backward()
 
+    def test_mark_non_differentiable_none(self):
+        # This used to segfault because MyFunction would send back null
+        # gradients to MulBackward
```

Comment:
Can you mention that it's related to C++ functions not creating grads that are undefined? It would work fine if MulBackward was a Python function